### PR TITLE
*: add leak detection for tests

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Currently, we don't have any leak related tests.

### What is changed and how it works?
This PR adds leak detection for our tests and does some cleanup. But for now, `api` package is not enabled due to some leak problems. We need further investigation.
This PR can be reviewed after #1758 is merged.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pkg/testutil/leak.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	goleak.IgnoreTopFunction("github.com/syndtr/goleveldb/leveldb.(*DB).compactionError"),
+	goleak.IgnoreTopFunction("google.golang.org/grpc.(*ccBalancerWrapper).watcher"),
+	goleak.IgnoreTopFunction("google.golang.org/grpc.(*ccResolverWrapper).watcher"),
+	goleak.IgnoreTopFunction("google.golang.org/grpc.(*addrConn).createTransport"),
+	goleak.IgnoreTopFunction("go.etcd.io/etcd/pkg/logutil.(*MergeLogger).outputLoop"),
+	// TODO: remove the below options once we fixed the http connection leak problems
+	goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+	goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+	goleak.IgnoreTopFunction("net/http.(*persistConn).readLoop"),
+	goleak.IgnoreTopFunction("runtime.goparkunlock"),
```
Comment (disksing) [inline pkg/testutil/leak.go]: Is it caused by we do not handle http response body correctly?

Comment (rleungx) [reply pkg/testutil/leak.go]: I'm not sure yet, still need investigation.

**Comment by rleungx** [comment]
PTAL @lhy1024 @nolouch @disksing

File: server/api/etcd_api_test.go [hunk 2]
```diff
@@ -15,21 +15,16 @@ package api
...
 
-type testEtcdAPISuite struct {
-	hc *http.Client
-}
+type testEtcdAPISuite struct{}
 
-func (s *testEtcdAPISuite) SetUpSuite(c *C) {
-	s.hc = newHTTPClient()
-}
+func (s *testEtcdAPISuite) SetUpSuite(c *C) {}
```
Comment (lhy1024) [inline server/api/etcd_api_test.go]: why do we need a empty function?

File: server/server_test.go [hunk 3]
```diff
@@ -161,6 +167,15 @@ func (s *testServerSuite) TestCheckClusterID(c *C) {
 	cfgA.InitialCluster = originInitial
 	svr, err := CreateServer(cfgA, nil)
 	c.Assert(err, IsNil)
-	err = svr.Run(ctx)
+
+	etcd, err := embed.StartEtcd(svr.etcdCfg)
```
Comment (lhy1024) [inline server/server_test.go]: why to add it?

Comment (rleungx) [reply server/server_test.go]: Here only needs to run the etcd server.

**Pull request merged**